src/ImageRunModal.jsx

```diff
@@ -27,13 +27,112 @@ const units = {
+                                     parseInt(port.hostPort) >= 1 && parseInt(port.hostPort) <= 65535);
+        this.props.onChange(validated.map(port => port.hostPort + ':' + port.containerPort + '/' + port.protocol));
+    }
+
+    render () {
+        const dialogValues = this.state;
+        return (
+            <div>
+                {
+                    dialogValues.publish.map((port, idx) => {
+                        return (
+                            <div className='publish-port-form ct-form-layout' key={'published-port-' + idx}>
+                                <div role='group' className='ct-form-layout-split'>
+                                    <input className='form-control'
+                                           type='number'
+                                           step={1}
+                                           min={1}
+                                           max={65535}
+                                           placeholder={_("container port")}
+                                           value={port.containerPort || ''}
+                                           onChange={e => this.onPortChange(idx, 'containerPort', e.target.value)} />
+                                    <Select.Select className='form-control'
+                                                   initial={port.protocol}
+                                                   onChange={value => this.onPortChange(idx, 'protocol', value)} >
+                                        <Select.SelectEntry data='TCP' key='TCP'>
+                                            {_("TCP")}
+                                        </Select.SelectEntry>
+                                        <Select.SelectEntry data='UDP' key='UDP'>
+                                            {_("UDP")}
+                                        </Select.SelectEntry>
+                                    </Select.Select>
+                                    <input className='form-control'
+                                           type='number'
+                                           step={1}
+                                           min={1}
+                                           max={65535}
+                                           placeholder={_("host port")}
+                                           value={port.hostPort || ''}
+                                           onChange={e => this.onPortChange(idx, 'hostPort', e.target.value)} />
+                                </div>
+                                <div role='group' className='ct-form-layout-split'>
+                                    <Button className='btn btn-default pficon-close'
+                                            disabled={dialogValues.publish.length === 1 &&
+                                                      (dialogValues.publish[0]['containerPort'] === null ||
```

Comment:
=== null can be replaced with !dialogValue.publish[0]['containerPort'], I think it looks better.